### PR TITLE
Adding VIN Validation

### PR DESCRIPTION
**Pull request by dberny**
When a user submits their entry for a VIN, the app will display text underneath with a warning if it is not a valid VIN according to current standards.

### CODE REVIEW

File: components/textFields/AdvancedOpenTextField.js
Comment (xieruishen) [inline components/textFields/AdvancedOpenTextField.js:79]: it seems like when i console log data.maxLength, it is always undefined so the length check doesn't seem to do anything?

File: components/textFields/AdvancedOpenTextField.js
Comment (xieruishen) [inline components/textFields/AdvancedOpenTextField.js:213]: I wonder if instead of choosing one of the invalid states if (both are true), should we display all invalidity of the VIN?

File: components/textFields/AdvancedOpenTextField.js
Comment (xieruishen) [inline components/textFields/AdvancedOpenTextField.js:223]: I think this text might need to a different color like red as currently it is not very distinctive. A user could potentially saw the bar green turned green after entering and submitting some value to the text field and miss this text warning thinking the VIN was fine.

**Review by xieruishen — CHANGES_REQUESTED** [review]
Overall looks great! The vin validation part is good. Just some smaller things in the comments:
- data.maxLength seems to be undefined all the time so maybe just take out that part since we only care about if the Vin is valid or not
- if we do figure out the correct data.maxlength check, maybe display both warnings if what user entered is both an invalid vin and violate length rule?
- the text warning prob should be a red error color to be more distinctive.
- make sure to pull from the latest dev branch with the major changes on the homescreen before merge. 

:)

File: components/textFields/AdvancedOpenTextField.js
Comment (dberny) [reply components/textFields/AdvancedOpenTextField.js:223]: Changed to red, but I wonder if it should stop the bar from turning green? There is a chance they have an old VIN (pre 1980 i think) that gets the warning but should still work

File: components/textFields/AdvancedOpenTextField.js
Comment (dberny) [reply components/textFields/AdvancedOpenTextField.js:79]: the length check is there in case a question sets that value (although I don't think any do), and was created last year. Should we just remove it or leave that in case we want it in the future?

Comment (xieruishen) [reply components/textFields/AdvancedOpenTextField.js:79]: i guess we can leave it and add some comments documenting the current status for the future :)

File: components/textFields/AdvancedOpenTextField.js
Comment (xieruishen) [reply components/textFields/AdvancedOpenTextField.js:223]: sounds good! Yeah we prob don't worry about stopping the bar from turning green now since if the bar doesn't turn green, whatever they entered even if they intend it that way wouldn't be saved by the app redux state

**Comment by xieruishen** [comment]
looks good to merge if we just add some comments for the weird length check situation :)

**Pull request merged**
